### PR TITLE
goToDefinition: find only the value if it's the RHS of an assignment

### LINKED ISSUE #33520: Duplicate definitions for simple file that uses module.exports
From https://github.com/microsoft/vscode/issues/80678

**TypeScript Version:**  3.7.0-dev.20190919

**Search Terms:**
- 

**Repo**
For the JavaScript:
```js
module.exports.test = async () => {
    return 123;
}

module.exports.test();
```

Run `go to definition` on `test` in `test()`

**Expected behavior:**
It returns a single definition for the function

**Actual behavior:**
Two definitions are returned

```
[Trace  - 6:36:34 PM] <semantic> Sending request: definitionAndBoundSpan (260). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/san/m1.js",
    "line": 5,
    "offset": 17
}
[Trace  - 6:36:34 PM] <semantic> Response received: definitionAndBoundSpan (260). Request took 2 ms. Success: true 
Result: {
    "definitions": [
        {
            "file": "/Users/matb/projects/san/m1.js",
            "start": {
                "line": 1,
                "offset": 16
            },
            "end": {
                "line": 1,
                "offset": 20
            },
            "contextStart": {
                "line": 1,
                "offset": 1
            },
            "contextEnd": {
                "line": 1,
                "offset": 20
            }
        },
        {
            "file": "/Users/matb/projects/san/m1.js",
            "start": {
                "line": 1,
                "offset": 23
            },
            "end": {
                "line": 3,
                "offset": 2
            }
        }
    ],
    "textSpan": {
        "start": {
            "line": 5,
            "offset": 16
        },
        "end": {
            "line": 5,
            "offset": 20
        }
    }
}
```

**Related Issues:** 
- #33428

### PR DESCRIPTION
**Pull request by elibarzilay**
Fixes #33520

### CODE REVIEW

File: tests/cases/fourslash/goToDefinitionVariableAssignment.ts [hunk 1]
```diff
@@ -4,9 +4,9 @@
 // @checkJs: true
 // @filename: foo.js
 ////const Bar;
-////const /*def1*/Foo = /*def2*/Bar = function () {}
+////const Foo = /*def*/Bar = function () {}
```
Comment (sheetalkamat) [inline tests/cases/fourslash/goToDefinitionVariableAssignment.ts:7]: I am not sure this is correct. I would expect this to go to `function () {}` instead.

Comment (elibarzilay) [reply tests/cases/fourslash/goToDefinitionVariableAssignment.ts:7]: I noticed that too, but this is how it was before, and it's not in the code that I changed, so unrelated to the fix...  (And BTW, there's some special treatment for `function` there which is doing it -- it would get the function if it was an arrow function.)

File: src/services/goToDefinition.ts [hunk 2]
```diff
@@ -246,7 +249,9 @@ namespace ts.GoToDefinition {
         // There are cases when you extend a function by adding properties to it afterwards,
         // we want to strip those extra properties.
         // For deduping purposes, we also want to exclude any declarationNodes if provided.
-        const filteredDeclarations = filter(symbol.declarations, d => d !== declarationNode && (!isAssignmentDeclaration(d) || d === symbol.valueDeclaration)) || undefined;
+        const filteredDeclarations =
```
Comment (RyanCavanaugh) [inline src/services/goToDefinition.ts:252]: Is this just a whitespace change?

File: src/services/goToDefinition.ts [hunk 3]
```diff
@@ -330,15 +335,11 @@ namespace ts.GoToDefinition {
 
     /** Returns a CallLikeExpression where `node` is the target being invoked. */
     function getAncestorCallLikeExpression(node: Node): CallLikeExpression | undefined {
-        const target = climbPastManyPropertyAccesses(node);
-        const callLike = target.parent;
+        const target = findAncestor(node, n => !isRightSideOfPropertyAccess(n));
```
Comment (RyanCavanaugh) [inline src/services/goToDefinition.ts:338]: Is this related to the bug?

File: tests/cases/fourslash/goToDefinitionVariableAssignment.ts (hunk 1 above)
Comment (sheetalkamat) [reply tests/cases/fourslash/goToDefinitionVariableAssignment.ts:7]: But its changing behavior of getting one definition instead of two.. It would be confusing to go to Bar here and not Foo when doing new Foo() ?

Comment (elibarzilay) [reply tests/cases/fourslash/goToDefinitionVariableAssignment.ts:7]: It's "half of a change", since already it *did* point at `Bar =` before.

In any case, I agree that it would be better to point at the expression instead, but that would be a change in a different place in the code.  There is one part that finds the value, and then a different part checks whether to show just the value or also the declared names.  I changed the latter, and what you're talking about is in the former which I didn't change.  So that should be a different PR, I think.

File: src/services/goToDefinition.ts (hunk 2 above)
Comment (elibarzilay) [reply src/services/goToDefinition.ts:252]: Yeah, sorry, late night push, and I forgot to add a comment about this just breaking the long line.  I'll remove it if you prefer not touching other parts.

File: src/services/goToDefinition.ts (hunk 3 above)
Comment (elibarzilay) [reply src/services/goToDefinition.ts:338]: Also a leftover when I was trying a different approach, and thought that it might be good to leave it.  (Same as above, I'll remove it if preferred.)

File: tests/cases/fourslash/goToDefinitionVariableAssignment.ts (hunk 1 above)
Comment (elibarzilay) [reply tests/cases/fourslash/goToDefinitionVariableAssignment.ts:7]: I'll open a new issue for this.
